substrate/frame/support/src/traits/tokens/fungible/union_of.rs

```diff
@@ -115,80 +115,80 @@ impl<
 	type AssetId = AssetKind;
 	type Balance = Left::Balance;
 
-	fn total_issuance(asset: Self::AssetId) -> Self::Balance {
-		match Criterion::convert(asset) {
+	fn total_issuance(asset: &Self::AssetId) -> Self::Balance {
+		match Criterion::convert(asset.clone()) {
```

Comment:
Hmm I wonder if `Criterion::convert` should also take a ref?

Reply:
I initially tried, but got lost in the sea of errors. There are so many generic types and trait bounds used here, hard to follow.

I will give it another try.

Reply:
Can't make it work.. got as far as https://github.com/dianpopa/polkadot-sdk/commit/090c74efa31d71604124ed3e660aba74e288d944, maybe someone can help or do it in another PR.